kratos/utilities/svd_utils.h

```diff
@@ -144,73 +149,75 @@ class SVDUtils
     {
         const SizeType m = InputMatrix.size1();
         const SizeType n = InputMatrix.size2();
-        
-        if(SMatrix.size1() != m || SMatrix.size2() != n)
-        {
+
+        if(SMatrix.size1() != m || SMatrix.size2() != n) {
             SMatrix.resize(m, n, false);
         }
-        SMatrix = InputMatrix;
-        
-        if(UMatrix.size1() != m || UMatrix.size2() != m)
-        {
+        noalias(SMatrix) = InputMatrix;
+
+        if(UMatrix.size1() != m || UMatrix.size2() != m) {
             UMatrix.resize(m, m, false);
         }
         noalias(UMatrix) = IdentityMatrix(m);
-        
-        if(VMatrix.size1() != n || VMatrix.size2() != n)
-        {
+
+        if(VMatrix.size1() != n || VMatrix.size2() != n) {
             VMatrix.resize(n, n, false);
         }
         noalias(VMatrix) = IdentityMatrix(n);
-        
-        const TDataType relative_tolerance = Tolerance * SparseSpaceType::TwoNorm(InputMatrix);
-        
-        std::size_t iter = 0;
-        
-        while (LocalSpaceType::JacobiNorm(SMatrix) > relative_tolerance)
-        {
-            for (SizeType i = 0; i < n; i++)
-            {
-                for (SizeType j = i+1; j < n; j++)
-                {
+
+        const TDataType relative_tolerance = Tolerance * LocalSpaceType::TwoNorm(InputMatrix);
+
+        IndexType iter = 0;
+
+        // We create the auxuliar matrices (for aliased operations)
+        MatrixType auxiliar_matrix_mn(m, n);
+        MatrixType auxiliar_matrix_m(m, m);
+        MatrixType auxiliar_matrix_n(n, n);
+
+        // We compute Jacobi
+        while (LocalSpaceType::JacobiNorm(SMatrix) > relative_tolerance) {
+            for (IndexType i = 0; i < n; i++) {
+                for (IndexType j = i+1; j < n; j++) {
                     MatrixType j1(m, m);
                     MatrixType j2(n, n);
-                    
+
                     Jacobi(j1, j2, SMatrix, m, n, i, j);
-                    
+
                     const MatrixType aux_matrix = prod(SMatrix, j2);
                     noalias(SMatrix) = prod(j1, aux_matrix);
-                    UMatrix = prod(UMatrix, trans(j1));
-                    VMatrix = prod(trans(j2), VMatrix);
+                    noalias(auxiliar_matrix_m) = prod(UMatrix, trans(j1));
```

Comment:
@jcotela @loumalouomega I am currently also looking into the required AMatrix changes and need some help
First question: Here you introduce an aux_matrix bcs in this case you actually DO have aliasing and AMatrix assumes noalias by default right?

Second question: Wouldn't it make more sense then to have an `alias` operation in AMatrix in case you have aliasing? Having to create an aux-object seems less efficient to me and also its more code

Thanks for your help

Reply:
I agree with that

Reply:
@pooyan-dadvand what do you think?